Automattic/jetpack
# JITM: Show ToS update notice

#### Changes proposed in this Pull Request:
* Allows JITMs to trigger AJAX actions when the CTA button is clicked.
* Adds an AJAX action for the ToS JITM to mark the ToS as accepted.

![image](https://user-images.githubusercontent.com/1233880/73758526-0178fb00-476b-11ea-80a6-d474a59a56fd.png)

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
Enhances the JITMs so they can display legal updates that need to be accepted such as the ToS 2020 update notice. See p58i-8v7-p2

#### Testing instructions:
- Apply D38880-code to your WP.com sandbox.
- Make sure you don't have ToS marked as accepted. See D38880-code for instructions on how to reset ToS acceptance.
- [Spin up a JN site running this branch](https://jurassic.ninja/create/?jetpack-beta&branch=add/jitm-ajax).
- Set up Jetpack and connect the JN site to WP.com.
- Open the Blog RC and add the `tos-test` blog sticker to the JN site.
- Go to WP Admin > Settings > Jetpack Constants.
- Set `JETPACK__SANDBOX_DOMAIN` to the URL of your WP.com sandbox.
- Access any WP Admin page of your Jetpack test site.
- You should see a ToS update notice.
- Click on Accept.
- Make sure the JITM is hidden and the ToS are accepted.
- Reload the WP Amin page.
- Make sure the ToS update notice doesn't show up.
- Reset ToS acceptance and repeat steps with a [JN site running the master branch](https://jurassic.ninja/create/?jetpack-beta).
- Sandbox jetpack.com and wordpress.com.
- Since the AJAX handler is not available, clicking on the CTA button should perform a fallback redirect:
  - CTA links to `jetpack.com/redirect?source=jitm-tos`
  - That link redirects to `wordpress.com/tos/accept` which marks ToS as accepted
- Make sure you're redirected back to the same WP Admin where you saw the JITM.
- Make sure the ToS update notice doesn't show up.

#### Proposed changelog entry for your changes:
JITM: Show ToS update notice
